### ISSUE TITLE
SchemaDiff returns inconsistent response if there is no difference

### ISSUE THREAD

**Issue by StrategieLukas** [directus/directus#24260]
### Describe the Bug

When I call the schemaDiff-Endpoint from the SDK, I receive a 204 HTTP Response object instead of the intended data object.

When there is a difference in schema (i.e. one field in this case) and I log the output of `client.request(schemaDiff(schema))`, I get:

```
{
  hash: '85560b835305ac552770ddd4742869c1e074a6',
  diff: { collections: [], fields: [ [Object] ], relations: [] }
}
```

When there is no difference, I instead receive an HTTP Reponse object:

```
Response {
  [Symbol(realm)]: null,
  [Symbol(state)]: {
    aborted: false,
    rangeRequested: false,
    timingAllowPassed: true,
    requestIncludesCredentials: true,
    type: 'default',
    status: 204,
    timingInfo: {
       ...
    },
    cacheState: '',
    statusText: 'No Content',
    headersList: HeadersList {
      cookies: null,
      [Symbol(headers map)]: [Map],
      [Symbol(headers map sorted)]: null
    },
    urlList: [ [URL] ],
    body: null
  },
  [Symbol(headers)]: HeadersList {
    cookies: null,
    [Symbol(headers map)]: Map(9) {
      'content-security-policy' => [Object],
      'x-powered-by' => [Object],
      'vary' => [Object],
      'access-control-allow-credentials' => [Object],
      'access-control-expose-headers' => [Object],
      'cache-control' => [Object],
      'date' => [Object],
      'connection' => [Object],
      'keep-alive' => [Object]
    },
    [Symbol(headers map sorted)]: null
  }
}
```

This makes it inconvenient to work with the result, because I first need to check if it is a diff-object or an http response code. Previously, this was not an issue. The problem occurred after these version updates:

Directus: 10.5.x -> 11.3.5
SDK: 11.0.3 -> 18.0.3

I found a similar issue, but it seems to be a different problem there: https://github.com/directus/directus/issues/17987

### To Reproduce

1. Call `client.request(schemaDiff(schema))` from the directusSDK with an unchanged schema and log the result
2. Observe that the result is the http message described above

### Directus Version

v11.3.5

### Hosting Strategy

Self-Hosted (Docker Image)

### Workaround
I can check if the response has a "diff" field to differentiate between the intended objects and the http response

**Comment by ComfortablyCoding** [comment]
Thanks for the report, I can confirm the inconsistency. Potentially what we should do is remove the hash and return the diff with empty arrays for its respective keys.

P.S. I opened #24276 (an re-linked the PR) for the SDK returning the response object for a 204 response as that is an unrelated issue.

**Comment by ComfortablyCoding** [comment]
Circling back here, as there is no diff a 204 response makes sense in this case as opposed to checking through the empty diff. The SDK issue will be fixed via #24276
